From e69243fb04c0b94ee0db72c639dbcb55755be957 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 21 Jun 2023 14:47:53 +0200
Subject: [PATCH 0489/1141] arm64: dts: st: use nvmem cells to store resource
 table in stm32mp251.dtsi

The stm32_rproc driver gets the resources tables size and address in
tamp register thanks to the NVMEM API.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Id033a152c030c296adbee3e52426f74e775dc8f1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/326926
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 5316e7e2dac5..7534f02e63da 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2346,6 +2346,12 @@ nvram: nvram@46010100 {
 				boot_mode: tamp-bkp@180 {
 					reg = <0x180 0x4>;
 				};
+				rsc_tbl_addr: tamp-bkp@184 {
+					reg = <0x184 0x4>;
+				};
+				rsc_tbl_size: tamp-bkp@188 {
+					reg = <0x188 0x4>;
+				};
 			};
 
 			reboot_mode: reboot-mode {
@@ -2638,6 +2644,8 @@ m33_rproc: m33@a040000 {
 			st,syscfg-cm-state = <&pwr 0x204 0x0000000c>;
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+			nvmem-cells = <&rsc_tbl_addr>, <&rsc_tbl_size>;
+			nvmem-cell-names = "rsc-tbl-addr", "rsc-tbl-size";
 
 			status = "disabled";
 		};
-- 
2.39.2

